src/main/java/com/google/sps/servlets/LoadInterviewsServlet.java

```diff
@@ -0,0 +1,256 @@
+        "Offset greater than 720 minutes (12 hours): %s",
+        timezoneOffsetMinutes);
+    ZoneOffset timezoneOffset = convertIntToOffset(timezoneOffsetMinutes);
+    ZonedDateTime day = generateDay(instant, timezoneOffset);
+    ZonedDateTime utcTime = day.withZoneSameInstant(ZoneOffset.UTC);
+    // TODO: Decide how precise this range should be. Must start after or at now, when should end
+    // be?
+    TimeRange range =
+        new TimeRange(utcTime.toInstant(), utcTime.toInstant().plus(6, ChronoUnit.DAYS));
+    List<PossibleInterviewSlot> possibleInterviews =
+        getPossibleInterviewSlots(range, timezoneOffset);
+
+    String date = possibleInterviews.isEmpty() ? "" : possibleInterviews.get(0).date();
+    List<ArrayList<PossibleInterviewSlot>> possibleInterviewsForWeek =
+        new ArrayList<ArrayList<PossibleInterviewSlot>>();
+
+    if (!possibleInterviews.isEmpty()) {
+      ArrayList<PossibleInterviewSlot> dayOfSlots = new ArrayList<PossibleInterviewSlot>();
+      for (PossibleInterviewSlot possibleInterview : possibleInterviews) {
+        if (!possibleInterview.date().equals(date)) {
+          possibleInterviewsForWeek.add(dayOfSlots);
+          dayOfSlots = new ArrayList<PossibleInterviewSlot>();
+          date = possibleInterview.date();
+        }
+        dayOfSlots.add(possibleInterview);
+      }
+      possibleInterviewsForWeek.add(dayOfSlots);
+    }
+
+    request.setAttribute("weekList", possibleInterviewsForWeek);
+    RequestDispatcher rd = request.getRequestDispatcher("/possibleInterviewTimes.jsp");
+
+    try {
+      rd.forward(request, response);
+    } catch (ServletException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  // Uses an Instant and a timezoneOffset to create a ZonedDateTime instance.
+  private static ZonedDateTime generateDay(Instant instant, ZoneOffset timezoneOffset) {
+    return instant.atZone(ZoneId.ofOffset("UTC", timezoneOffset));
+  }
+
+  // This method takes the timezoneOffsetMinutes int and converts it
+  // into a proper ZoneOffset instance.
+  private static ZoneOffset convertIntToOffset(int timezoneOffsetMinutes) {
+    return ZoneOffset.ofHoursMinutes((timezoneOffsetMinutes / 60), (timezoneOffsetMinutes % 60));
+  }
+
+  private List<PossibleInterviewSlot> getPossibleInterviewSlots(
+      TimeRange range, ZoneOffset timezoneOffset) {
+
+    Set<String> interviewers = availabilityDao.getUsersAvailableInRange(range.start(), range.end());
+    Set<PossibleInterviewSlot> possibleInterviews = new HashSet<PossibleInterviewSlot>();
+
+    // We don't want to schedule an interview for a user with themself, so we are removing
+    // the current user's id from the list.
+    UserService userService = UserServiceFactory.getUserService();
+    String userEmail = userService.getCurrentUser().getEmail();
+    String userId = userService.getCurrentUser().getUserId();
+
+    // Since UserId does not have a valid Mock, if the id is null (as when testing), it will be
```

Comment:
I know this comment has been copy pasted all over the code base (yuck) but it doesn't actually make sense.

The real cause is that "Users returned from the LocalUserService (in tests) do not have userIds".

Reply:
Done